src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -0,0 +1,289 @@
+const simpleGit = require("simple-git");
+const fs = require("fs");
+const glob = require("glob");
+const path = require("path");
+
+const jsdom = require("jsdom");
+const { JSDOM } = jsdom;
+const {
+  createAmplifyBranches,
+  startReleaseJob,
+  createAmplifyApp,
+  readBuildSpec,
+} = require("./amplifyUtils");
+const BRANCH_NAME = "chore-amplify-migration-change-permalinks";
+const ORGANIZATION_NAME = "isomerpages";
+
+interface AmplifyAppInfo {
+  appId: string;
+  repoName: string;
+  name: string;
+  repoPath: string;
+}
+
+const listOfRepos: [string, string][] = [
+  ["repo-name", "Human Friendly Name"], // Modify this line when running the script
+];
+
+async function main() {
+  for (const [repoName, name] of listOfRepos) {
+    try {
+      await migrateRepo(repoName, name);
+    } catch (e) {
+      console.error(e);
+    }
+  }
+}
+
+async function migrateRepo(repoName: string, name: string) {
+  const homeDir = process.env.HOME;
+  const repoPath = `${homeDir}/isomer-migrations/${repoName}`;
+
+  try {
+    const build_spec = await readBuildSpec();
+    const appId = await createAmplifyApp(repoName, build_spec);
+    const amplifyAppInfo: AmplifyAppInfo = {
+      appId,
+      repoName,
+      name,
+      repoPath,
+    };
+    await createAmplifyBranches(amplifyAppInfo);
+    await startReleaseJob(amplifyAppInfo);
+    await modifyRepo(amplifyAppInfo);
+    await pushChangesToRemote(amplifyAppInfo);
+    await generateSqlCommands(amplifyAppInfo);
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+async function modifyRepo({ repoName, appId, repoPath }: AmplifyAppInfo) {
+  if (fs.existsSync(repoPath)) {
+    console.log(
+      `Repository ${repoName} already exists. Pulling changes from origin.`
+    );
+    await simpleGit(repoPath).pull("origin", "staging");
+  } else {
+    console.log(`Cloning ${repoName} repository from ${ORGANIZATION_NAME}...`);
+    await simpleGit().clone(
+      `https://github.com/${ORGANIZATION_NAME}/${repoName}.git`,
+      repoPath
+    );
+  }
+
+  // get list of branches
+  const branches = await simpleGit(repoPath).branchLocal();
```

Comment:
good point! I do pull right before this operation, ie 
if repo exists locally, pull origin
if not, clone (effectively referring to upsteam)

Reply:
9d09a84
Realised my earlier explanation is wrong. The reason why I don't need to pull from origin is because I don't intend to push this branch to remote! Added comment to reflect this.